Readme.md

```diff
@@ -0,0 +1,1432 @@
+```
+
+This feature affects type inference across parameters list and in the case of case classes it excludes all members of the non-first parameter lists from the `equals`, `hashCode` and `unapply` implementations.
+
+The latter point especially very often causes code which relies on multiple parameter lists to be confusing. Therefore, using them in general is forbidden.
+
+There are two use cases for which multiple parameters lists are allowed. The first one is definition of implicit arguments (see the respective section above):
+
+```scala
+def doSomeWork(arg: String)(implicit ec: ExecutionContext) = ...
+```
+
+The second one is when the second parameter list is used to accept a single functional value:
+
+```scala
+def run(x: Int)(f: Int => String): String = f(x)
+
+run(10) { x => x.toString }
+```
+
+Multiple parameter lists here allow using more concise syntax for passing an anonymous function which increases readability. That said, the need to declare such functions in itself should be rare, because of the rule against DSLs.
+
+The second use case, however, does not extend to multiple parameter lists which accept functions; in other words, something like the following is not allowed:
+
+```scala
+def computeAnswer(onComplete: Int => Unit)
+                 (onFailure: Throwable => Unit) = ...
+
+computeAnswer { x => println(x) } { e => e.printStackTrace() }
+```
+
+If you do need to create a function which, for example, accepts multiple callbacks, use named parameters instead:
+
+```scala
+def computeAnswer(onComplete: Int => Unit,
+                  onFailure: Throwable => Unit) = ...
+
+run(
+  onComplete = { x => println(x) },
+  onFailure = { e => e.printStackTrace }
+)
+```
+
+## Symbolic methods and infix and postfix notation
+
+Scala allows to define methods which consist of symbols instead of regular letters. Combined with  the ability to call any method with the infix notation this allows one to define custom operators:
+
+```scala
+class Connectable {
+  def =+~>(other: Connectable): Connectable = ...
+}
+
+val x = new Connectable
+val y = new Connectable
+x =+~> y
+```
+
+Symbolic method names do not convey what they do, and with the exception of regular well-known symbols (mathematical operations, `++`/`--` on collections, boolean operators, negation, etc.) tend to be obscure by themselves: `a =+~> x` could mean absolutely anything.
+
+Therefore it is prohibited to define custom symbolic methods, while their use with the 3rd-party libraries should be as limited as possible. Naturally, using unicode characters, which is also allowed by the language, strictly prohibited as well.
+
+Additionally, using the infix notation for calling regular, non-symbolic methods is discouraged and could be used only if it significantly increases readability. One example of a sufficiently motivated use of infix notation are matchers and other parts of the tests definition in ScalaTest.
```

Comment:
I disagree. It is true that infix notation for collection combinators may reduce some of the syntactic noise, but it is bad for three reasons.

First, infix operators usually suggest some kind of symmetry of its arguments - either in mathematical sense (like commutativity of `+` for numbers) or just in common sense (it is natural for collections to be joined together with `++`). An exception from this basic rule are operators which join only one element to a collection (like `+` on sets), but these are actually natural extensions of the symmetry rule - they still mean "addition" to collection, and addition is naturally represented by an infix operator.

However, there is no such symmetry in the slightest in the case of collection operations. Most of them accept a function as their argument, and functions and collections are not symmetric in any conceivable sense. Moreover, in cases like ScalaTest the symmetry rule is also broken completely, but the infix notation still helps to make the code readable like a natural language. On the other hand, in the case of collection operators this does not hold, because their names do not help with natural-like readability at all.

Second, using infix notation with multiple chained operations (and since in the infix case the argument tend to be quite small, it is very tempting to chain several operations), I dare say, one of the worst cases of the unintelligibility:

```scala
coll map someTransformation filter selectBySomething takeWhile isValid reduce combine
// versus
coll.map(someTransformation)
  .filter(selectBySomething)
  .takeWhile(isValid)
  .reduce(combine)
```

If one of the functions passed as arguments are lambdas instead of named functions, the first variant becomes even worse, while the second one retains its relative clarity.

And finally, when all usages of a certain method are uniform, e.g. when a particular method is always called in the same way, it helps with readability with *much* greater effect than local reduction of syntactic noise.